shared/chat/conversations-list/container.js

```diff
@@ -1,138 +1,37 @@
 // @flow
+import * as I from 'immutable'
 import ConversationList from './index'
-import React, {Component} from 'react'
 import {connect} from 'react-redux'
-import {formatTimeForConversationList} from '../../util/timestamp'
-import {globalColors} from '../../styles'
-import {loadInbox, selectConversation, newChat} from '../../actions/chat'
-import {participantFilter, newestConversationIDKey} from '../../constants/chat'
-import {List} from 'immutable'
+import {createSelectorCreator, defaultMemoize} from 'reselect'
+import {loadInbox, newChat} from '../../actions/chat'
 
-import type {ConversationIDKey, InboxState, SupersededByState} from '../../constants/chat'
 import type {TypedState} from '../../constants/reducer'
 
-let _loaded = false
+const getInbox = (state: TypedState) => state.chat.get('inbox')
+const getSupersededByState = (state: TypedState) => state.chat.get('supersededByState')
+const getAlwaysShow = (state: TypedState) => state.chat.get('alwaysShow')
+const getPending = (state: TypedState) => state.chat.get('pendingConversations')
 
-class ConversationListContainer extends Component {
-  componentWillMount () {
-    if (!_loaded) {
-      _loaded = true
-      this.props.loadInbox()
-    }
-  }
-
-  _derivedProps (rekeyInfo, unreadCount, isSelected) {
-    // Derived props
-    const youNeedToRekey = rekeyInfo && !rekeyInfo.get('rekeyParticipants').count() && rekeyInfo.get('youCanRekey')
-    const participantNeedToRekey = rekeyInfo && !!rekeyInfo.get('rekeyParticipants').count()
-    const hasUnread = !!unreadCount
-    const subColor = isSelected ? globalColors.black_40 : hasUnread ? globalColors.white : globalColors.blue3_40
-    const showBold = !isSelected && hasUnread
-    const backgroundColor = isSelected ? globalColors.white : hasUnread ? globalColors.darkBlue : globalColors.darkBlue4
-    const usernameColor = isSelected ? globalColors.black_75 : hasUnread ? globalColors.white : globalColors.blue3_60
-
-    return {
-      backgroundColor,
-      hasUnread,
-      participantNeedToRekey,
-      showBold,
-      subColor,
-      usernameColor,
-      youNeedToRekey,
-    }
-  }
-
-  render () {
-    const pendingRows = this.props.pending.map((users, conversationIDKey) => {
-      const unreadCount = 0
-      const participants = participantFilter(users, this.props.you)
-      const isSelected = this.props.selectedConversation === conversationIDKey
-      const isMuted = false
-      const rekeyInfo = null
-      const timestamp = formatTimeForConversationList(Date.now(), this.props.nowOverride)
-      const snippet = ''
-      const onSelectConversation = this.props.onSelectConversation
-
-      return {
-        conversationIDKey,
-        isMuted,
-        isSelected,
-        onSelectConversation,
-        participants,
-        rekeyInfo,
-        snippet,
-        timestamp,
-        unreadCount,
-      }
-    }).toList()
+const createImmutableEqualSelector = createSelectorCreator(defaultMemoize, I.is)
```

Comment:
so really this pass was to touch the store and whatnot very little. we could keep these things unfiltered in the store and filter them in the reselect, then it would pass.